src/Symfony/Component/Serializer/Deserialize/Splitter/JsonLexer.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Deserialize\Splitter;
+
+use Symfony\Component\Serializer\Exception\InvalidResourceException;
+
+/**
+ * @author Mathias Arlaud <mathias.arlaud@gmail.com>
+ *
+ * @experimental in 7.0
+ */
+final class JsonLexer
```

Comment:
Is it a port of an existing JSON lexer or a totally new algorithm (in which case we will have to double-check its security and performance)?

Reply:
This lexer is a brand-new algorithm (inspired by existing ones).

That's why it is indeed tested over the https://github.com/nst/JSONTestSuite to make sure that it is compliant with the RFC 8259 (see https://github.com/symfony/symfony/pull/51718/files#diff-267a936d2e81b79f6fcdfd66d56380cfbaaff6656b60177998eeeac77be8614aR47)